support/fc_sort.py

```diff
@@ -0,0 +1,153 @@
+        """
+        self.path = path
+        self.file_type = file_type
+        self.context = context
+
+        self.compute_diffdata()
+
+    def compute_diffdata(self):
+        """ Compute the interal values needed for comparing two file context definitions
+        """
+
+        self.meta = False
+        self.stem_len = 0
+        self.str_len = 0
+
+        skip_escaped = False
+
+        for char in self.path:
+            if skip_escaped:
+                skip_escaped = False
+                continue
+
+            if char in ('.', '^', '$', '?', '*', '+', '|', '[', '(', '{',):
+                self.meta = True
+            if char == '\\':
+                skip_escaped = True
+
+            if not self.meta:
+                self.stem_len += 1
+
+            self.str_len += 1
+
+
+def compare_fcds(a, b):
+    """ Compare two file context definitions
+
+    Returns:
+      -1 if a is less specific than b
+       0 if a and be are equally specific
+       1 if a is more specific than b
+    The comparison is based on the following statements,
+    in order from most important to least important, given a and b:
+       If a is a regular expression and b is not,
+        -> a is less specific than b.
+       If a's stem length is shorter than b's stem length,
+        -> a is less specific than b.
+       If a's string length is shorter than b's string length,
+        -> a is less specific than b.
+       If a does not have a specified type and b does,
+        -> a is less specific than b.
+    """
+
+    # Check to see if either a or b have meta characters and the other doesn't
+    if a.meta and not b.meta:
+        return -1
+    if b.meta and not a.meta:
+        return 1
+
+    # Check to see if either a or b have a shorter stem length than the other
+    if a.stem_len < b.stem_len:
+        return -1
+    if b.stem_len < a.stem_len:
+        return 1
+
+    # Check to see if either a or b have a shorter string length than the other
+    if a.str_len < b.str_len:
+        return -1
+    if b.str_len < a.str_len:
+        return 1
+
+    # Check to see if either a or b has a specified type and the other doesn't
+    if not a.file_type and b.file_type:
+        return -1
+    if not b.file_type and a.file_type:
+        return 1
+
+    # If none of the above conditions were satisfied, then a and b are equally specific
+    return 0
+
+
+def main(argv=None):
+
+    parser = argparse.ArgumentParser(description='Sort file context definitions')
+    parser.add_argument('infile', metavar='INFILE', nargs=1, type=Path,
+                        help='input file of the original file context definitions')
+    parser.add_argument('outfile', metavar='OUTFILE', nargs='?', type=Path, default=None,
+                        help='output file for the sorted file context definitions')
+    args = parser.parse_args(argv)
+
+    file_context_definitions = []
+
+    # Parse the input file
+    with args.infile[0].open('r') as fd:
```

Comment:
If you remove the nargs=1 then you can just do `args.infile`.

Reply:
done